### ISSUE TITLE
ActiveModel::ForbiddenAttributesError

### ISSUE THREAD

**Issue by kulisu** [securityroots/dradispro-tracker#324]
- ~~/opt/dradispro/dradispro/current/app/controllers/attachments_controller.rb~~
- /opt/dradispro/dradispro/current/app/controllers/password_resets_controller.rb
- app/controllers/dradis/pro/issuelib/admin/issuelib/entries_controller.rb
- .. etc ?

**Comment by etdsoft** [comment]
Hi @kulisu thanks for this.

How do you trigger the error on:

`./app/controllers/attachments_controller.rb`

and

`./app/controllers/password_resets_controller.rb`

This seems like it could be related to the Rails 4 migration but I haven't been able to reproduce.


I suspect the 3rd one is some sort of typo? The .gem file has nothing to do with ActiveModel controllers.

**Comment by kulisu** [comment]
Hello @etdsoft,

thanks for your quick replay.
it's seems that some controllers didnt migrate to Rails4.

i'm so sorry that `attachments_controller.rb` is a typo and it's related to #325 issue.
`password_resets_controller.rb`, `entries_controller.rb` are still using old-style to update attributes.

and how to reproduce these ?
just click the reset password link, submit a new password and you will get a 500 server error.
and the error will happen when you trying to update an issue in IssueLibrary lol.

```ruby
# File: password_resets_controller.rb
# Line: 19,47

    elsif !params[:user][:password].blank? && @user.update_attributes(params[:user])

# File: entries_controller.rb
# Line: 20,10

      if @entry.update_attributes(params[:entry])
```

```ruby
# monkey patch temporarily here :D

# File: config/initializers/z_99_PATCH_EntriesController.rb
if @entry.update_attributes(entry_params)

private
  def entry_params
    # dont forget :state, we found that we cant update issue's state.
    params.require(:entry).permit(:content, :state)
  end
end

# File: config/initializers/z_99_PATCH_PasswordResetsController.rb
elsif !params[:user][:password].blank? && @user.update_attributes(user_params)

private
  def user_params
    params.require(:user).permit(:password, :password_confirmation)
  end
```

**Comment by etdsoft** [comment]
There is a new version of IssueLibrary add-on that it's compatible with Rails 4, please get in touch with Support to get it.

Confirmed bug in PasswordResetsController, upon changing the password. I had checked the creation of the password reset, and that seemed to have worked before.

**Comment by etdsoft** [comment]
Fixed, ships in v2.3
